xdsl/dialects/arm_neon.py

```diff
@@ -345,6 +345,54 @@ def assembly_line_args(self):
         )
 
 
+@irdl_op_definition
+class DSVecMovOp(ARMInstruction):
+    """
+    Variant of MOV instruction which extracts a value from a specified lane in a NEON register into a general-purpose register.
+    e.g. MOV X0, V3.S[1]
+    Set X0 to the value of the second single word (bits 32-63) in V3.
+    This instruction is an alias of UMOV.
+    """
+
+    name = "arm_neon.dsvec.mov"
+    s = operand_def(NEONRegisterType)
+    d = result_def(IntRegisterType)
+    scalar_idx = prop_def(IntegerAttr[i8])
+    arrangement = prop_def(NeonArrangementAttr)
+    assembly_format = (
+        "$s `[` $scalar_idx `]` $arrangement attr-dict `:` type($s) `->` type($d)"
+    )
+
+    def __init__(
+        self,
+        s: Operation | SSAValue,
+        *,
+        d: IntRegisterType,
+        arrangement: NeonArrangement | NeonArrangementAttr,
+        comment: str | StringAttr | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+        if isinstance(arrangement, NeonArrangement):
+            arrangement = NeonArrangementAttr(arrangement)
+        super().__init__(
+            operands=(s,),
+            attributes={
+                "comment": comment,
+                "arrangement": arrangement,
```

Comment:
This will need to go in the properties dictionary, there also seems to be no way to specify a `scalar_idx`?